File: 401. binary-watch.java
class Solution {
    public List<String> readBinaryWatch(int turnedOn) {
        List<String> result = new ArrayList<>();
        
        for (int hour = 0; hour < 12; hour++) {
            
            for (int minute = 0; minute < 60; minute++) {
                
                int totalBits = Integer.bitCount(hour) + Integer.bitCount(minute);
                
                if (totalBits == turnedOn) {
                    
                    String time = hour + ":" + 
                                  (minute < 10 ? "0" + minute : minute);
                    
                    result.add(time);
                }
            }
        }
        
        return result;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        int turnedOn1 = 1;
        List<String> output1 = sol.readBinaryWatch(turnedOn1);
        System.out.println("Example 1: Input: turnedOn = " + turnedOn1);
        System.out.println("Output: " + output1);
        System.out.println("Expected: [0:01, 0:02, 0:04, 0:08, 0:16, 0:32, 1:00, 2:00, 4:00, 8:00]");
        System.out.println();
        
        // Example 2:
        int turnedOn2 = 9;
        List<String> output2 = sol.readBinaryWatch(turnedOn2);
        System.out.println("Example 2: Input: turnedOn = " + turnedOn2);
        System.out.println("Output: " + output2);
        System.out.println("Expected: []");
    }
}

File: 401. binary-watch.py
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> list[str]:

        if turnedOn > 8: return []
        ans = []
        
        for hour in range(12):
            for minute in range(60):
                if hour.bit_count() + minute.bit_count() == turnedOn:
                    ans.append(f"{hour}:{minute:02d}")

        return ans

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    turnedOn = 1
    output = sol.readBinaryWatch(turnedOn)
    print(f"Example 1: Input: turnedOn = {turnedOn}")
    print(f"Output: {output}")
    print(f"Expected: ['0:01','0:02','0:04','0:08','0:16','0:32','1:00','2:00','4:00','8:00']")
    print()
    
    # Example 2:
    turnedOn = 9
    output = sol.readBinaryWatch(turnedOn)
    print(f"Example 2: Input: turnedOn = {turnedOn}")
    print(f"Output: {output}")
    print(f"Expected: []")
        